packages/protvista-proteomics-adapter/src/ProtvistaProteomicsAdapter.js

```diff
@@ -1,20 +1,83 @@
 import ProtvistaFeatureAdapter, {
   renameProperties,
-  formatTooltip
+  formatTooltip,
 } from "protvista-feature-adapter";
 import { v1 } from "uuid";
 
-export const transformData = data => {
+const proteomicsTrackProperties = (feature) => {
+  return {
+    category: "PROTEOMICS",
+    type: feature.unique ? "unique" : "non_unique",
+    tooltipContent: formatTooltip(feature),
+    protvistaFeatureId: v1(),
+  };
+};
+
+export const transformData = (data) => {
   let adaptedData = [];
+
   if (data && data.length !== 0) {
-    adaptedData = data.features.map(feature => {
-      return Object.assign(feature, {
-        category: "PROTEOMICS",
-        type: feature.unique ? "unique" : "non_unique",
-        tooltipContent: formatTooltip(feature),
-        protvistaFeatureId: v1()
-      });
+    // To merge PTM data present in same residue in a same length peptide, have a map [key: start-end-phospho site 1-... phosphosite n, value: corresponding feature elements]
+    const ptmMap = {};
+    data.features.forEach((feature) => {
+      let ft = `${feature.begin}-${feature.end}`;
+      if (feature.ptms) {
+        feature.ptms.forEach((ptm) => {
+          ft += `-${ptm.position}`;
+        });
+        if (ptmMap[ft]) {
+          ptmMap[ft] = [...ptmMap[ft], feature];
+          ptmMap[ft].push(feature);
+        } else {
+          ptmMap[ft] = [feature];
+        }
```

Comment:
```suggestion
        ptmMap[ft] = ft in ptmMap ? [...ptmMap[ft], feature] : [feature]
```
I find ternaries quite useful in these situations